scipy/optimize/_linprog_highs.py

```diff
@@ -0,0 +1,411 @@
+            success : bool
+                ``True`` when the algorithm succeeds in finding an optimal
+                solution.
+            status : int
+                An integer representing the exit status of the algorithm.
+
+                ``0`` : Optimization terminated successfully.
+
+                ``1`` : Iteration or time limit reached.
+
+                ``2`` : Problem appears to be infeasible.
+
+                ``3`` : Problem appears to be unbounded.
+
+                ``4`` : The HiGHS solver ran into a problem.
+
+            nit : int
+                The total number of iterations performed.
+                For ``solver='simplex'``, this includes iterations in all
+                phases. For ``solver='ipm'``, this does not include
+                crossover iterations.
+            crossover_nit : int
+                The number of primal/dual pushes performed during the
+                crossover routine for ``solver='ipm'``.  This is ``0``
+                for ``solver='simplex'``.
+            message : str
+                A string descriptor of the exit status of the algorithm.
+
+    """
+
+    _check_unknown_options(unknown_options)
+
+    # Catch message_level warnings that HiGHS currently misses:
+    message_level = (
+        _check_invalid_option_values(message_level, 'message_level',
+                                     {None, 0, 1, 2, 3, 4, 5, 6, 7}, MESSAGE_LEVEL_MINIMAL))
+
+    # SIMPLEX_STRATEGY_PRIMAL (4) is experimental -- disallow use!
+    simplex_strategy = (
+        _check_invalid_option_values(simplex_strategy, 'simplex_strategy',
+                                     {None, 0, 1, 2, 3}, 1))
+
+    statuses = {
+        MODEL_STATUS_NOTSET: (4, 'HiGHS Status Code 0: HighsModelStatusNOTSET'),
+        MODEL_STATUS_LOAD_ERROR: (4, 'HiGHS Status Code 1: HighsModelStatusLOAD_ERROR'),
+        MODEL_STATUS_MODEL_ERROR: (2, 'HiGHS Status Code 2: HighsModelStatusMODEL_ERROR'),
+        MODEL_STATUS_MODEL_EMPTY: (4, 'HiGHS Status Code 3: HighsModelStatusMODEL_EMPTY'),
+        MODEL_STATUS_PRESOLVE_ERROR: (4, 'HiGHS Status Code 4: HighsModelStatusPRESOLVE_ERROR'),
+        MODEL_STATUS_SOLVE_ERROR: (4, 'HiGHS Status Code 5: HighsModelStatusSOLVE_ERROR'),
+        MODEL_STATUS_POSTSOLVE_ERROR: (4, 'HiGHS Status Code 6: HighsModelStatusPOSTSOLVE_ERROR'),
+        MODEL_STATUS_RDOVUB: (4, 'HiGHS Status Code 10: HighsModelStatusREACHED_DUAL_OBJECTIVE_VALUE_UPPER_BOUND'),
+        MODEL_STATUS_PRIMAL_INFEASIBLE: (2, "The problem is infeasible."),
+        MODEL_STATUS_PRIMAL_UNBOUNDED: (3, "The problem is unbounded."),
+        MODEL_STATUS_OPTIMAL: (0, "Optimization terminated successfully."),
+        MODEL_STATUS_REACHED_TIME_LIMIT: (1, "Time limit reached."),
+        MODEL_STATUS_REACHED_ITERATION_LIMIT: (1, "Iteration limit reached."),
+    }
+
+    c, A_ub, b_ub, A_eq, b_eq, bounds, x0 = lp
+
+    lb, ub = bounds.T.copy()                # separate bounds, copy->C-cntgs
```

Comment:
These comments also trigger PEP8 issues and also to me they look a bit obvious. Do we need them?

Reply:
These comments are helpful to me because they explain why/how the inputs are changed got HiGHS compared to the other linprog methods.  For example, useful to know that transpose is done to keep `lb` and `ub` C-contiguous.